### ISSUE TITLE
[Question] Regression accuracy decrease after output scaling

### ISSUE THREAD

**Issue by rangerbottle** [cornellius-gp/gpytorch#1141]
Dear all,

I just used Gpytorch to make multivariate regression model. Code is here:

    class GPRegressionModel(gpytorch.models.ExactGP):
        def __init__(self, train_x, train_y, likelihood):
            super(GPRegressionModel, self).__init__(train_x, train_y, likelihood)
            self.mean_module = gpytorch.means.ConstantMean()
            self.covar_module = ScaleKernel(gpytorch.kernels.RBFKernel())
       
        def forward(self, x):
            mean_x = self.mean_module(x)
            covar_x = self.covar_module(x)
            return gpytorch.distributions.MultivariateNormal(mean_x, covar_x)

    def GP_Regression(X_train, y_train, X_test):
         likelihood = gpytorch.likelihoods.GaussianLikelihood().cuda()
         model = GPRegressionModel(X_train, y_train, likelihood).cuda()
         model.train()
         likelihood.train()
         optimizer = torch.optim.Adam(model.parameters(), lr=0.01)
         mll = gpytorch.mlls.ExactMarginalLogLikelihood(likelihood, model)
         training_iterations = 200
         def train():
               for i in range(training_iterations):
                     optimizer.zero_grad()
                     output = model(X_train)
                     loss = -mll(output, y_train)
                     loss.backward()
                     optimizer.step()
         train()
         model.eval()
         likelihood.eval()
         with torch.no_grad():
                observed_pred = likelihood(model(X_test))
                mean = observed_pred.mean.cpu().numpy()
         return model, mean

The interesting thing is, after I used z-score to scale output (y), I found the accuracy of test data is decreased.
Before the scaling the mean absolute error (MAE) is 0.000728545024613147  but after scaling this value increase to 0.0008450164360690002  

I tried same data with XGBoost, also comparing with output scale/ output non-scale, the MAE is not changed too much, as 0.000797721806100908 / 0.0007888684071199087

I tried to increase the training iteration to get smaller loss, but it will make the test data's accuracy even lower, maybe due to overfitting.

So is there any clue about this question?
torch 1.3.1
gpytorch 1.0.1

Thanks in advance

**Comment by gpleiss** [comment]
After z-scoring, a MAE of 0.0008450164360690002 is very low (random guessing gives a MAE of 1). This shouldn't be a practical difference at all. My hunch is that - if your problem had larger errors (e.g. a MAE in the 0.01 or the 0.1 range) there would be even less of a relative difference.

I think the 0.0001 difference is likely due to floating-point errors. GPyTorch uses single precision for faster computation. If you really care about the extra 0.0001 then try calling `model.double(), X_train.double(), y_train.double(), X_test.double()`.

**Comment by rangerbottle** [comment]
> a MAE
Thanks for your comment, 0.0008450164360690002 is the result after de-normalized the prediction comparing with true value. The original output value is [-0.001, 0.003]. 
I have tried with double, but result is not improve too much.
Another thing I observed is the loss.
Before the output scaling, about 200 iterations training, the loss is around -0.117
After the output scaling, same iterations, the loss is around 0.887, then I increase the iterations tuning the learning rate, and found the 0.303 is the lowest loss, but the accuracy is worse at this time, I think it could be overfiiting.
so I just wandering, is there any relationship between the loss and prediction accuracy?
What is the best loss for regression?
Thannks a lot

**Comment by gpleiss** [comment]
Training loss and test-time accuracy should ideally be related - however it's not always the case. It sounds to me like you're overfitting to your training data (which - though less likely with GPs than parametric methods - is still possible). This might be because your test data differs sufficiently from you training data.

The loss values also depend on how your data is scaled. If your y values are in [-0.001, 0.003] you'll get much lower loss values than if your ys are scaled between [-1, 1].
